lib/src/chart/radar_chart/radar_chart_data.dart

```diff
@@ -0,0 +1,267 @@
+import 'dart:ui';
+
+import 'package:equatable/equatable.dart';
+import 'package:fl_chart/fl_chart.dart';
+import 'package:fl_chart/src/utils/lerp.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+
+typedef GetTitleByIndexFunction = String Function(int index);
+
+class RadarChartData extends BaseChartData {
+  //ToDo(payam) : document data classes
+  RadarChartData({
+    this.dataSets = const [],
+    this.fillColor = Colors.transparent,
+    this.chartBorderData = const BorderSide(color: Colors.black, width: 2),
+    this.radarTouchData,
+    this.getTitle,
+    @required this.titleCount,
+    this.titleTextStyle = const TextStyle(color: Colors.black, fontSize: 12),
+    this.titlePositionPercentageOffset = 0.2,
+    @required this.tickCount,
+    this.ticksTextStyle = const TextStyle(color: Colors.black, fontSize: 9),
+    this.tickBorderData = const BorderSide(color: Colors.black, width: 2),
+    this.gridBorderData = const BorderSide(color: Colors.black, width: 2),
+    FlBorderData borderData,
+  })  : assert(dataSets != null, "the dataSets field can't be null"),
+        assert(titleCount != null && titleCount >= 2, "RadarChart need's more then 2 titles"),
+        assert(tickCount != null && tickCount >= 2, "RadarChart need's more then 2 ticks"),
+        assert(
+          titlePositionPercentageOffset >= 0 && titlePositionPercentageOffset <= 1,
+          'titlePositionPercentageOffset must be something between 0 and 1 ',
+        ),
+        assert(
+          dataSets.firstWhere(
+                (element) => element.dataEntries.length != titleCount,
+                orElse: () => null,
+              ) ==
+              null,
+          'dataSets value count must be equal to titleCount value',
+        ),
+        super(borderData: borderData, touchData: radarTouchData);
+
+  final List<RadarDataSet> dataSets;
+
+  final Color fillColor;
+  final BorderSide chartBorderData;
+
+  final GetTitleByIndexFunction getTitle;
+  final int titleCount;
+  final TextStyle titleTextStyle;
+
+  /// the [titlePositionPercentageOffset] is the place of showing title on the section
+  /// the degree is statically on the center of each section,
+  /// but the radius of drawing is depend of this field,
+  /// this field should be between 0 and 1,
+  /// if it is 0 the title will be drawn near the inside section,
+  /// if it is 1 the title will be drawn near the outside of section,
+  /// the default value is 0.5, means it draw on the center of section.
+  final double titlePositionPercentageOffset;
```

Comment:
I didn't understand what it does. and nothing happened when I changed it.

Reply:
it works similarly to `PieChartSectionData.titlePositionPercentageOffset`. this is the padding between chart and titles.